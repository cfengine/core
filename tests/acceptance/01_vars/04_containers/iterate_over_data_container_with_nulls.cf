body common control
{
        inputs => { "../../default.cf.sub" };
        bundlesequence  => { default("$(this.promise_filename)") };
        version => "1.0";
}

bundle agent init
{
  commands:
    windows::
      "$(G.dos2unix) $(this.promise_filename).expected" -> { "ENT-10433" };

  files:
      "$(G.testdir)/actual.txt"
        delete => tidy;
}

bundle agent test
{
  meta:
      "description" -> { "CFE-2230" }
        string => "Test that iteration over the getindices() of a data container
                   is not interrupted by nulls.";

  vars:
      "data"            data => '{
  "instanceId": "i-f444554e",
  "billingProducts": null,
  "instanceType": "m3.large",
  "accountId": "444444444444",
  "pendingTime": "2015-12-02T19:47:57Z",
  "imageId": "ami-44444444",
  "kernelId": "aki-44444444",
  "ramdiskId": null,
  "architecture": "x86_64",
  "region": "eu-west-1",
  "version": "2010-08-31",
  "availabilityZone": "eu-west-1c",
  "privateIp": "10.44.44.4",
  "devpayProductCodes": null
}';

      "data2"           data => '{
  "instanceId" : "i-f444554e",
  "instanceType" : "m3.large",
  "accountId" : "444444444444",
  "pendingTime" : "2015-12-02T19:47:57Z",
  "imageId" : "ami-44444444",
  "kernelId" : "aki-44444444",
  "architecture" : "x86_64",
  "region" : "eu-west-1",
  "version" : "2010-08-31",
  "availabilityZone" : "eu-west-1c",
  "privateIp" : "10.44.44.4",
}';

      "keys" slist => getindices("data");
      "$(keys)" string => "$(data[$(keys)])";

      "keys2" slist => getindices("data2");
      "$(keys2)" string => "$(data2[$(keys2)])";

  reports:

      "with nulls: $(keys) = $(data[$(keys)])"
        report_to_file => "$(G.testdir)/actual.txt";

      "without nulls: $(keys2) = $(data2[$(keys2)])"
        report_to_file => "$(G.testdir)/actual.txt";
}

bundle agent check
{
  methods:
      "Pass/Fail"
        usebundle => dcs_check_diff( "$(G.testdir)/actual.txt",
                                     "$(this.promise_filename).expected",
                                     $(this.promise_filename) );
}
