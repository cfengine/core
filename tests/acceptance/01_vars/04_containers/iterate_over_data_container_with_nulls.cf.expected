with nulls: instanceId = i-f444554e
with nulls: instanceType = m3.large
with nulls: accountId = 444444444444
with nulls: pendingTime = 2015-12-02T19:47:57Z
with nulls: imageId = ami-44444444
with nulls: kernelId = aki-44444444
with nulls: architecture = x86_64
with nulls: region = eu-west-1
with nulls: version = 2010-08-31
with nulls: availabilityZone = eu-west-1c
with nulls: privateIp = 10.44.44.4
without nulls: instanceId = i-f444554e
without nulls: instanceType = m3.large
without nulls: accountId = 444444444444
without nulls: pendingTime = 2015-12-02T19:47:57Z
without nulls: imageId = ami-44444444
without nulls: kernelId = aki-44444444
without nulls: architecture = x86_64
without nulls: region = eu-west-1
without nulls: version = 2010-08-31
without nulls: availabilityZone = eu-west-1c
without nulls: privateIp = 10.44.44.4
